homeassistant/components/tuya/climate.py

```diff
@@ -203,8 +206,10 @@ def __init__(
         # Determine HVAC modes
         self._attr_hvac_modes: list[HVACMode] = []
         self._hvac_to_tuya = {}
-        if enum_type := self.find_dpcode(
-            DPCode.MODE, dptype=DPType.ENUM, prefer_function=True
+        if (not self._is_thermostat) and (
```

Comment:
This assumes the MODE only applies to the `wk` category, but I highly doubt that is the case. For example, `rs` category also has modes.

Reply:
actually, my idea was quite opposite - at least in my thermostat, which fall into this wk category, I don't see any point to mess with modes. Modes from Tuya device are "auto" and "manual", while Climate device expect Heat and Cool. So, with my changes I'm trying to eliminate any dependencies on mode that may came from device and "hardcode" only Heat and Off mode in terms of Climate device.

Reply:
>  at least in my thermostat, which fall into this wk category, I don't see any point to mess with modes. 

That sounds like an off-standard issue, as this mode is documented.
I'm happy to improve support, but your current change is breaking for existing devices as pointed out in my comments.

Reply:
My change should not affect any device in rs category, that's the main point of adding this "flag" variable self._is_thermostat which is set only for wk category.
And according to the documentation, mode for my heater device is declared as enum with range ["auto", "manual"]. I don't see how this can be used in terms of Climate device hvac_mode? Of course, if I'm wrong and there are other devices in wk category that has mode with values, meaningful for hvac_mode, I can suggest to shorten definition of _is_thermostat flag to only my particular device model, in order to leave others devices intact.

Reply:
Perhaps another approach here will be to check device modes and if they are "auto" and "manual", then add a single "heat" mode to hvac_modes?

Reply:
There is no certainty it is a heating device.

Reply:
I don't actually know how to make it better. But with current implementation, using Device.Turn_On and Turn_Off services, I wasn't able to reliably control my thermostat. Also, Climate card displays a confusing set of modes, like OFF, HEAT and HEAT. And when you select any of those HEAT, both become selected. I will think how this can be improved.

Reply:
I've reworked my changes here. Difference with original code is only to avoid duplication of same mode more than once